index.js

```diff
@@ -1,38 +1,202 @@
+
+const player1ShipCount = document.querySelector("#ships_player1");
+const player2ShipCount = document.querySelector("#ships_player2");
+
+//Players' Names
+player1.name = prompt("Player #1 type your name");
+const player1Label = document.querySelector("#name_player1");
+player1Label.textContent = player1.name;
+player2.name = prompt("Player #2 type your name");
+const player2Label = document.querySelector("#name_player2");
+player2Label.textContent = player2.name;
+
+//*****************Randomly Add Ships to each Board*****************
+//alert("Your ships are being put on alert... Be ready for the battle");
+
+const boardSetup = (player) => {
+  for (let i = 0; player.shipCount < 4; i++) {
+    let x = Math.floor(Math.random() * 4);
+    let y = Math.floor(Math.random() * 4);
+    if (player.gameBoard[x][y] === 1) {
+      continue;
+    }
+    player.shipCount++;
+    player.gameBoard[x][y] = 1;
+  }
+  return player.gameBoard;
+};
+boardSetup(player1);
+boardSetup(player2);
+
+//***************Set up a beginner.*****************
+//coin toss: 50/50
+function setBeginner() {
+  let coin = Math.floor(Math.random() * 2 + 1);
+  if (coin === 1) {
+    currentPlayer = player1;
+    opponent = player2;
+    activeBoard = board_player2;
+    inactiveBoard = board_player1;
+    player1ShipCount.textContent = currentPlayer.shipCount;
+    player2ShipCount.textContent = opponent.shipCount;
+  } else if (coin !== 1) {
+    currentPlayer = player2;
+    opponent = player1;
+    activeBoard = board_player1;
+    inactiveBoard = board_player2;
+    player2ShipCount.textContent = currentPlayer.shipCount;
+    player1ShipCount.textContent = opponent.shipCount;
+  }
+
+  turnLabel.textContent = currentPlayer.name;
+}
+setBeginner();
+
+console.log(opponent.gameBoard); //=========================debugging
+console.log(currentPlayer.gameBoard);
+
+//******************The main game flow**********************
+const launchGame = (player) => {
+  for (var x = 0; x < 4; x++) {
+    const li = document.createElement("li"); // creating childs for the list (board), in this case represent a row number 'x' of the board
+
+    for (var y = 0; y < 4; y++) {
+      const cell = document.createElement("div");
+      cell.className = "square"; // adding css properties to make it looks like a square
+      cell.textContent = `${x},${y}`; // saves the coordinates as a string value 'x,y'
+
+      cell.value = player.gameBoard[x][y]; //state of the cell
+
+      //this function adds the click event to each cell
+      cell.addEventListener("click", (e) => {
+        if (activeBoard.contains(cell)) {
+          //validation of the board
+
+          [currentPlayer, opponent] = [opponent, currentPlayer]; // switch players and boards
+          [activeBoard, inactiveBoard] = [inactiveBoard, activeBoard];
+
+          turnLabel.textContent = currentPlayer.name;
+
+          let cell = e.target; // get the element clicked
+
+          if (cell.value === 1) {

```

Comment:
after clicking the cell , the cell is still available to click. after a  hit, player can click same cell to win  the game, we can set a cell value to "0" and make it disable to be clicked again